src/types/fieldExtractor.ts

```diff
@@ -1,13 +1,25 @@
-import { getTrait, NFT } from './nft';
+import { getTrait, NFT, NftFactory } from './nft';
 
-export enum CustomFieldExtractor {
+export type TitleExtractor = (nft: NFT) => string;
+
+export enum CustomTitleExtractor {
   METADATA_NAME = 'metadata.name',
   ATTRIBUTES_TRAIT_SONG_TITLE = 'attributes.trait.songTitle',
-  ATTRIBUTES_TRAIT_TRACK = 'attributes.trait.track'
+  ATTRIBUTES_TRAIT_TRACK = 'attributes.trait.track',
+  DEFAULT = ATTRIBUTES_TRAIT_TRACK
 }
 
-export const fieldExtractors: any = {
+export const titleExtractors: any = {
```

Comment:
if this is just for titles, then i think we can use a specific type since it should always be nft => string?